dali/kernels/imgproc/resample/resampling_impl.cuh

```diff
@@ -203,94 +334,379 @@ __device__ void ResampleVert_Channels(
+ *
+ * The function fills the output in block-sized horizontal/depthwise spans.
+ * Filter coefficients are pre-calculated for each span to avoid recalculating them for each
+ * column, and stored in a shared memory block.
+ */
+template <int static_channels = -1, typename Dst, typename Src>
+__device__ void ResampleVert_Channels(
+    ivec3 lo, ivec3 hi,
+    float src_y0, float scale,
+    Dst *__restrict__ out, ptrdiff_vec<2> out_strides,
+    const Src *__restrict__ in, ptrdiff_vec<2> in_strides, ivec3 in_shape, int dynamic_channels,
+    ResamplingFilter filter, int support) {
+  using resample_shared::coeffs;
+
+  int out_stride_y = out_strides.x;
+  int in_stride_y = in_strides.x;
+  int out_stride_z = out_strides.y;
+  int in_stride_z = in_strides.y;
+  int in_h = in_shape.y;
+
+  src_y0 += 0.5f * scale - 0.5f - filter.anchor;
+
+  const float filter_step = filter.scale;
+
+  const bool huge_kernel = support > 256;
+  const int coeff_base = huge_kernel ? 0 : support*threadIdx.y;
+  const int channels = static_channels < 0 ? dynamic_channels : static_channels;
+
+  for (int i = lo.y; i < hi.y; i+=blockDim.y) {
+    int dy = i + threadIdx.y;
+    const float sy0f = dy * scale + src_y0;
+    const int sy0 = huge_kernel ? __float2int_rn(sy0f) : __float2int_ru(sy0f);
+    float f = (sy0 - sy0f) * filter_step;
+    __syncthreads();
+    if (huge_kernel) {
+      for (int k = threadIdx.x + blockDim.x*threadIdx.y; k < support; k += blockDim.x*blockDim.y) {
+        float flt = filter(f + k*filter_step);
+        coeffs[k] = flt;
+      }
+    } else {
+      for (int k = threadIdx.x; k < support; k += blockDim.x) {
+        float flt = filter(f + k*filter_step);
+        coeffs[coeff_base + k] = flt;
+      }
+    }
+    __syncthreads();
+
+    if (dy >= hi.y)
+      continue;
+
+    Dst *out_row = &out[dy * out_stride_y];
+
+    float norm = 0;
+    for (int k = 0; k < support; k++) {
+      norm += coeffs[coeff_base + k];
+    }
+    norm = 1.0f / norm;
+
+    for (int z = lo.z + threadIdx.z; z < hi.z; z += blockDim.z) {
+      Dst *out_slice = &out_row[z * out_stride_z];
+      const Src *in_slice = &in[z * in_stride_z];
+      for (int j = lo.x + threadIdx.x; j < hi.x; j += blockDim.x) {
+        Dst *out_col = &out_slice[j * channels];
+        const Src *in_col = &in_slice[j * channels];
+
+        if (static_channels < 0) {
+          for (int c = 0; c < channels; c++) {
+            float tmp = 0;
+
+            for (int k = 0; k < support; k++) {
+              int y = sy0 + k;
+              int ysample = y < 0 ? 0 : y >= in_h-1 ? in_h-1 : y;
+              float flt = coeffs[coeff_base + k];
+              Src px = __ldg(in_col + in_stride_y * ysample + c);
+              tmp = fmaf(px, flt, tmp);
+            }
+
+            out_col[c] = ConvertSat<Dst>(tmp * norm);
+          }
+        } else {
+          float tmp[static_channels < 0 ? 1 : static_channels];  // NOLINT - not a variable length array
```

Comment:
Again static_channels  is aleady >= 0.

Reply:
if the `if` was C++17 `if constexpr`, then yes.